pkg/services/mssql/common_update.go

```diff
@@ -0,0 +1,40 @@
+package mssql
+
+import (
+	"github.com/Azure/open-service-broker-azure/pkg/service"
+)
+
+func buildDatabaseUpdateGoTemplateParameters(
```

Comment:
iirc, in other modules we haven't differentiated between the Go template params that are fed into an initial ARM deployment vs those that are fed into an update to an existing ARM deployment. Here, the name seems to imply we're differentiating those two cases. I'm sure there is a reason for that. Can you explain for my own edification?

Reply:
The nature of this model (with different structure of params based on the plans) made it feel cleaner to implement in terms of interfaces, so in this case I needed to have a method that dealt with the update parameters from the instance vs the provision params. This was an attempt to make the methods very clearly spell out what they were working with. It is using the same template, so perhaps better naming might be:

_buildDatabaseGoTemplateParametersForUpdate_ and _buildDatabaseGoTemplateParametersForProvision_ to make it clear? 

The main difference is really: 
_p, err = details.getTierUpdateParameters(instance)_ vs _	p, err = details.getTierProvisionParameters(instance)_